src/forms/labels/labeled-field.js

```diff
@@ -88,23 +91,25 @@ function LabeledField({
   children,
   hideErrorLabel,
   label,
+  as: Component,
   ...rest
 }) {
   const { name } = input
   const { touched, invalid } = meta
   return (
-    <fieldset
-      className={classnames(className, {
+    <Component
+      className={classnames(className, 'field-wrapper', {
         error: hasInputError({ touched, invalid }),
         disabled: rest.disabled,
       })}
+      role="group"
```

Comment:
**aria role = group** MDN: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role

Does it make sense to assign `group` role to the container here? 

It is technically redundant when Component is `fieldset` whose implicit aria role is group (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#technical_summary)

This would be for when Component is `div`:
![Screen Shot 2023-09-25 at 11 39 51 AM](https://github.com/LaunchPadLab/lp-components/assets/55593736/afb19a92-ddaa-4de6-993d-79f98d55c3c8)
This role assignment would be necessary to pass the test that relies on role (if we are to keep the test as is): https://github.com/LaunchPadLab/lp-components/blob/main/test/forms/inputs/icon-input.test.js#L12

What about for individual checkbox/radio group? Is it odd that there is a `role=group` for the container that wraps the individual label and the corresponding checkbox?
![Screen Shot 2023-09-25 at 11 38 41 AM](https://github.com/LaunchPadLab/lp-components/assets/55593736/8bf8aade-02c5-4ac0-98df-b954512dc934)

cc: @chawes13

Reply:
Agreed. Since we are providing the ability now to set this to the correct element that would provide the correct aria role, I wouldn't think we'd need to specify one and should modify the test instead.

Reply:
Just a heads up, once we remove that role, several other tests will need to be updated.